saleor/graphql/product/types/channels.py

```diff
@@ -43,35 +47,81 @@ class Meta:
 
     @staticmethod
     def resolve_channel(root: models.ProductChannelListing, info, **_kwargs):
-        return ChannelByProductChannelListingIDLoader(info.context).load(root.id)
+        return ChannelByIdLoader(info.context).load(root.channel_id)
 
     @staticmethod
     @permission_required(ProductPermissions.MANAGE_PRODUCTS)
-    def resolve_purchase_cost(root: models.ProductChannelListing, *_args):
-        # TODO: Add dataloader.
-        variants = root.product.variants.all().values_list("id", flat=True)
-        channel_listings = models.ProductVariantChannelListing.objects.filter(
-            variant_id__in=variants, channel_id=root.channel_id
-        )
-        has_variants = True if len(variants) > 0 else False
-        purchase_cost, _ = get_product_costs_data(
-            channel_listings, has_variants, root.currency
+    def resolve_purchase_cost(root: models.ProductChannelListing, info, *_kwargs):
```

Comment:
Do you think about creating `resolvers.py` (if one does not exist) and moving there the logic of `resolve_purchase_cost` and `resolve_margin` to the corresponding functions?

Reply:
Also I'm thinking about creating separate data loaders instead of putting this logic into resolvers functions.

Reply:
After a discussion with @IKarbowiak, We decide to left the current solution for now.